src/nodemetadata.cpp

```diff
@@ -235,7 +236,8 @@ void NodeMetadataList::remove(v3s16 p)
 void NodeMetadataList::set(v3s16 p, NodeMetadata *d)
 {
 	remove(p);
-	m_data.insert(std::make_pair(p, d));
+	if (!d->empty())
+		m_data.insert(std::make_pair(p, d));
```

Comment:
```suggestion
		m_data.emplace(p, d);
```
might be pointless, who knows